CMSgov/design-system
# [WNMGDS-2625] Remove `core-js` polyfills for IE

## Summary

Fixes #2845 by removing `core-js` imports from the design system, which we only needed for IE support. [See the impact analysis here](https://github.com/CMSgov/design-system/discussions/2845#discussioncomment-7911136).

## How to test

I temporarily added bundle-analyzing capabilities to the build system and compared before and after. Either trust me or verify through some similar means. It's irradicated from our design system dependencies and bundles, even though it's still a dependency of Gatsby.

## Checklist
